parser.go

```diff
@@ -349,41 +384,148 @@ func (s *Session) readTemplateRecord(sl *slice) TemplateRecord {
 	return tr
 }
 
-func (s *Session) readVariableLength(sl *slice) (val []byte, err error) {
-	var l int
+func (s *Session) registerTemplateRecord(tr *TemplateRecord) {
+	if s.withTemplateIDAliasing {
+		tr.TemplateID = s.registerAliasedTemplateRecord(*tr)
+	} else {
+		s.registerUnaliasedTemplateRecord(*tr)
+	}
+}
 
-	l0 := sl.Uint8()
-	if l0 < 255 {
-		l = int(l0)
+func (s *Session) registerUnaliasedTemplateRecord(tr TemplateRecord) {
+	// Update templates and minimum record cache
+	tid := tr.TemplateID
+	specifiers := tr.FieldSpecifiers
+	minLength := calcMinRecLen(specifiers)
+	s.mut.Lock()
+	if minLength == 0 {
+		delete(s.specifiers, tid)
 	} else {
-		l = int(sl.Uint16())
+		s.specifiers[tid] = specifiers
 	}
+	s.minRecord[tid] = minLength
+	s.mut.Unlock()
+}
 
-	return sl.Cut(l), sl.Error()
+func (s *Session) registerAliasedTemplateRecord(tr TemplateRecord) uint16 {
+
+	var TemplateID uint16
+	if len(tr.FieldSpecifiers) == 0 {
+		TemplateID = s.withdrawTemplateRecord(tr)
+	} else {
+		TemplateID = s.aliasTemplateRecord(tr)
+	}
+
+	if debug {
+		dl.Printf("Mapped template id %d -> %d", tr.TemplateID, TemplateID)
+	}
+	return TemplateID
 }
 
-func (s *Session) registerTemplateRecord(tr TemplateRecord) {
-	// Update the template cache
-	tid := tr.TemplateID
+func (s *Session) aliasTemplateRecord(tr TemplateRecord) uint16 {
+	var buffer bytes.Buffer
+	binary.Write(&buffer, binary.BigEndian, tr.FieldSpecifiers)
+	hash := fmt.Sprintf("%x", sha1.Sum(buffer.Bytes()))
```

Comment:
```go
hash := hex.EncodeToString(sha1.Sum(buffer.Bytes())[:])
```

avoids pulling in the fmt dependency (which is large and currently not in here, otherwise I would not worry about it.)